core/src/main/java/de/danoeh/antennapod/core/service/FeedUpdateWorker.java

```diff
@@ -44,4 +89,51 @@ public Result doWork() {
 
         return Result.success();
     }
+
+    @Subscribe(sticky = true, threadMode = ThreadMode.MAIN)
```

Comment:
I think it is a bit strange doing this with an event. The worker could already be stopped (because refreshing is performed in a service, not the worker here) and could therefore not receive events when the download service is stopped. I think it would be better to create a new class `NewEpisodesNotification` (?) next to the `DownloadServiceNotification` class and call the method directly from `DownloadService`. Currently, that's just a broad direction. I am aware that it does not directly work like I suggested (storing state etc). Just a broad direction that I would find more clean and stable.